### PR TITLE
Add CHANGELOG.md

### PR DESCRIPTION
**Pull request by ralphtheninja**
Ok, here we go.

So maybe a bit overworked, but `cabal-core` is sort of like the flag ship module so might be worth the time to document all the versions. I've added a lot of small commits so it's easier to review each version separately and we can squash the PR later to make it a bit cleaner.

The workflow I use when making new release is basically:

```
npm run changelog:patch
# fill in details or tweak CHANGELOG.md
git add CHANGELOG.md && git commit -m 'Update changelog for x.y.z'
npm version patch
git push && git push --tags
npm publish
```

The first command will create a new section in the changelog (here showing patch as example). This will also populate the changelog with git commits, but if you want to add items manually you can also refer to pull requests e.g. `(#44)` and you can also refer to users on github by `(@substack, @cblgh)` and then `hallmark --fix` will automatically convert them to links.

### CODE REVIEW

**Comment by hackergrrl** [comment]
Excellent, thank you @ralphtheninja.

Does this mean there was no breaking protocol change created by the bump to `14.0.0`?

Also, would you mind adding the instructions for how to update the changelog to the README? Perhaps near the bottom in a section for developers?

**Comment by ralphtheninja** [comment]
> Does this mean there was no breaking protocol change created by the bump to `14.0.0`?

I don't feel I have the knowledge to answer this, but if you take a look at the diff the api didn't seem to break, https://github.com/cabal-club/cabal-core/compare/v13.2.0...v14.0.0 and I'm not sure why the new major version. Maybe due to updating `kappa-core`?



> Also, would you mind adding these helpful instructions for how to update the changelog to the README? Perhaps near the bottom in a section for developers?

Of course! Lets leave this PR hanging until then.

**Comment by ralphtheninja** [comment]
Updated docs and added table of contents as well.

**Comment by cblgh** [comment]
i don't remember why i made the major bump either >.<. looking at @ralphtheninja's comparison there was a lot going on though:
* hypercore-crypto major bump
* kappa-core major bump
* adding hyperswarm-web
* adding private messages

there's also some discussion in #104 that seems to have been on the side of breaking out into a new major, starting from https://github.com/cabal-club/cabal-core/pull/104#issuecomment-826375010 and its ensuing replies

---

this is so so nice @ralphtheninja, thank you so much for the tremendous effort & the [newly added how-to section](https://github.com/cabal-club/cabal-core/pull/107/commits/2b6b8f71532828b0d9a697a84dea46cd89e836a0) (!!)

thumbs up from me 👍 👍

**Comment by ralphtheninja** [comment]
@cblgh Seems we're pulling in `hypercore@9` even:

Screenshot from 2021-09-20 23-30-16

Let me update the changelog with this information, so we don't need to have this dicussion again :smile:

File: README.md [hunk 1]
```diff
@@ -2,6 +2,57 @@
 
 Core database, replication, swarming, and chat APIs for cabal.
 
+## Table of Contents
+
+<details><summary>Click to expand</summary>
```
Comment (ralphtheninja) [inline README.md:7]: Note that the table of contents is automatically generated by `hallmark` when doing `hallmark --fix`. So if any entries are added to the readme they should be automatically updated when the changelog is updated.

**Comment by hackergrrl** [comment]
Thank you for investigating where the breaking change was coming from @ralphtheninja.

Looks good to me also. Merging!

**Comment by ralphtheninja** [comment]
@hackergrrl You should really have squashed this :smile: Anyway, happy that you are happy with my contribution!

**Comment by hackergrrl** [comment]
@ralphtheninja I don't squash by default, but let me know next time & I'll give you time to squash it before we merge.

**Pull request merged**